src/main/routing/packet.c

```diff
@@ -86,29 +86,32 @@ const gchar* protocol_toString(ProtocolType type) {
     }
 }
 
-Packet* packet_new(PluginVirtualPtr payload, gsize payloadLength, guint hostID, guint64 packetID) {
+Packet* packet_new(Host* host) {
     Packet* packet = g_new0(Packet, 1);
     MAGIC_INIT(packet);
 
     packet->referenceCount = 1;
 
-    packet->hostID = hostID;
-    packet->packetID = packetID;
-
-    if (payload.val != 0 && payloadLength > 0) {
-        /* the payload starts with 1 ref, which we hold */
-        packet->payload = payload_new(payload, payloadLength);
-
-        /* application data needs a priority ordering for FIFO onto the wire */
-        packet->priority = host_getNextPacketPriority(worker_getActiveHost());
```

Comment:
It doesn't look like the priority is set anywhere in the new code.

Reply:
😱 good catch!